client/checkout/classic/upe-deferred-intent-creation/event-handlers.js

```diff
@@ -109,8 +109,55 @@ jQuery( function ( $ ) {
 		) {
 			for ( const upeElement of $( '.wcpay-upe-element' ).toArray() ) {
 				await mountStripePaymentElement( api, upeElement );
+				restrictPaymentMethodToLocation( upeElement );
 			}
 			maybeEnableStripeLink( api );
 		}
 	}
+
+	/**
+	 * Hides payment method if it has set specific countries in the PHP class.
+	 *
+	 * @param {Object} upeElement The selector of the DOM element of particular payment method to mount the UPE element to.
+	 **/
+	function restrictPaymentMethodToLocation( upeElement ) {
+		const paymentMethodsConfig = getUPEConfig( 'paymentMethodsConfig' );
+		const paymentMethodType = upeElement.dataset.paymentMethodType;
+		const isRestrictedInAnyCountry = !! paymentMethodsConfig[
+			paymentMethodType
+		].countries.length;
+
+		if ( isRestrictedInAnyCountry ) {
+			togglePaymentMethodForCountry(
+				paymentMethodType,
+				paymentMethodsConfig[ paymentMethodType ].countries
+			);
+			$( '#billing_country' ).on( 'change', function () {
+				togglePaymentMethodForCountry(
+					paymentMethodType,
+					paymentMethodsConfig[ paymentMethodType ].countries
+				);
+			} );
+		}
+	}
+
+	/**
+	 * @param {string} paymentMethodType The payment method type.
+	 * @param {Array}  supportedCountries A list of countries that given payment method is supported in.
+	 **/
+	function togglePaymentMethodForCountry(
+		paymentMethodType,
+		supportedCountries
+	) {
+		const billingCountry = document.querySelector( '#billing_country' )
```

Comment:
nitpick: In the file, we are using jQuery and vanilla JS to query the DOM elements, I'd stick with one

Reply:
Marking as resolved as per the issue with using vanilla JS mentioned in https://github.com/Automattic/woocommerce-payments/pull/7315#issuecomment-1746879451